### DISCUSSION TITLE
Optimizing GeoTiff tiles to be loaded in osgEarth

### DISCUSSION THREAD

**Discussion by shramir** [pelicanmapping/osgearth#1997] [Support Q&A] [answered]
I have a TMS (created by gdal2tiles.py) including thousands of GeoTiff files, containing the earth' elevation data, This TMS is correctly loaded in osgEarth but it renders the terrain too slow.

My Question is a step-by-step guideline optimizing GeoTiff tiles to be fastly loaded and rendered in osgEarth in a real time 3D application. Particularly, data type such as 16bit or 32 bit float/Int? Compression Types such as no compression, LZW , etc? setting GeoTiff overviews or Pyramids? Removing or preserving geotiff header? projection selection? selecting the image size 256 , 512,...? including No Data value(-32767) in files? and others tiff files options.

**Comment by gwaldron** [comment, marked as answer]
Hi. 
For general tips on optimizing GeoTIFFs, there is information here:
https://docs.osgearth.org/en/latest/data.html#optimizing-a-geotiff

However: osgEarth expects its elevation tiles to be set up a certain way. They needs to be 257x257 with overlap on the four sides. the `osgearth_conv` utility can generate them. Here's an example. First make a VRT from your many files using the GDAL `gdalbuildvrt` utility. Then:
```
osgearth_conv --in driver GDALElevation --in url myInputData.vrt
              --out driver TMSElevation --out url mytmsrepo/tms.xml
              --out format tif
```
Hope this helps.
